bundles/org.openhab.binding.matter/matter-server/src/client/ControllerNode.ts

```diff
@@ -0,0 +1,367 @@
+// Include this first to auto-register Crypto, Network and Time Node.js implementations
+import { CommissioningController } from "@project-chip/matter.js";
+import { NodeId } from "@matter/types";
+import { PairedNode, NodeStates, Endpoint } from "@project-chip/matter.js/device";
+import { Environment, Logger, StorageContext } from "@matter/general";
+import { ControllerStore } from "@matter/node";
+import { WebSocketSession } from "../app";
+import { EventType, NodeState } from '../MessageTypes';
+import { printError } from "../util/error";
+const logger = Logger.get("ControllerNode");
+
+/**
+ * This class contains represents the Matter Controller / Admin client
+ */
+export class ControllerNode {
+
+    private environment: Environment = Environment.default;
+    private storageContext?: StorageContext;
+    private nodes: Map<NodeId, PairedNode> = new Map();
+    commissioningController?: CommissioningController;
+
+    constructor(
+        private readonly storageLocation: string,
+        private readonly controllerName: string,
+        private readonly nodeNum: number,
+        private readonly ws: WebSocketSession,
+        private readonly netInterface?: string
+    ) { }
+
+    get Store() {
+        if (!this.storageContext) {
+            throw new Error("Storage uninitialized");
+        }
+        return this.storageContext;
+    }
+
+    /**
+     * Closes the controller node
+     */
+    async close() {
+        await this.commissioningController?.close();
+        this.nodes.clear();
+    }
+
+    /**
+     * Initializes the controller node
+     */
+    async initialize() {
+        const outputDir = this.storageLocation;
+        const id = `${this.controllerName}-${this.nodeNum.toString()}`
+        const prefix = "openHAB: ";
+        const fabricLabel = prefix + this.controllerName.substring(0, 31 - prefix.length);
+                
+        logger.info(`Storage location: ${outputDir} (Directory)`);
+        this.environment.vars.set('storage.path', outputDir)
+    
+        //TODO we may need to choose which network interface to use
```

Comment:
```suggestion
        // TODO: We may need to choose which network interface to use
```

Reply:
Apart from the format, we shouldn't forget about this README.

Reply:
Right now i don't have the ability for the user to limit which interfaces we bind to, i figured that would come in an update when needed, right now its not been a problem (but wanted a reminder of where it would go!)